dotnet/src/SemanticKernel/CoreSkills/TextMemorySkill.cs

```diff
@@ -37,49 +37,96 @@ public class TextMemorySkill
+        Verify.NotEmpty(key, "Memory key not defined");
+
+        context.Log.LogTrace("Recalling memory from collection '{0}'", collection);
+
+        var memory = await context.Memory.GetAsync(collection, key);
+        return memory != null ? memory.Text : string.Empty;
+    }
+
+    /// <summary>
+    /// Semantic search and return up to N memories related to the input text
     /// </summary>
     /// <example>
     /// SKContext["input"] = "what is the capital of France?"
     /// {{memory.recall $input }} => "Paris"
     /// </example>
-    /// <param name="ask">The information to retrieve</param>
-    /// <param name="context">Contains the 'collection' to search for information and 'relevance' score</param>
-    [SKFunction("Recall a fact from the long term memory")]
+    /// <param name="text">The input text to find related memories for</param>
+    /// <param name="context">Contains the 'collection' to search for the topic and 'relevance' score</param>
+    [SKFunction("Semantic search and return up to N memories related to the input text")]
     [SKFunctionName("Recall")]
-    [SKFunctionInput(Description = "The information to retrieve")]
-    [SKFunctionContextParameter(Name = CollectionParam, Description = "Memories collection where to search for information", DefaultValue = DefaultCollection)]
+    [SKFunctionInput(Description = "The input text to find related memories for")]
+    [SKFunctionContextParameter(Name = CollectionParam, Description = "Memories collection to search", DefaultValue = DefaultCollection)]
     [SKFunctionContextParameter(Name = RelevanceParam, Description = "The relevance score, from 0.0 to 1.0, where 1.0 means perfect match",
         DefaultValue = DefaultRelevance)]
-    public async Task<string> RecallAsync(string ask, SKContext context)
+    [SKFunctionContextParameter(Name = LimitParam, Description = "The maximum number of relevant memories to recall", DefaultValue = DefaultLimit)]
+    public string Recall(string text, SKContext context)
     {
         var collection = context.Variables.ContainsKey(CollectionParam) ? context[CollectionParam] : DefaultCollection;
-        Verify.NotEmpty(collection, "Memory collection not defined");
+        Verify.NotEmpty(collection, "Memories collection not defined");
 
         var relevance = context.Variables.ContainsKey(RelevanceParam) ? context[RelevanceParam] : DefaultRelevance;
         if (string.IsNullOrWhiteSpace(relevance)) { relevance = DefaultRelevance; }
 
-        context.Log.LogTrace("Searching memory for '{0}', collection '{1}', relevance '{2}'", ask, collection, relevance);
+        var limit = context.Variables.ContainsKey(LimitParam) ? context[LimitParam] : LimitParam;
+        if (string.IsNullOrWhiteSpace(relevance)) { relevance = DefaultLimit; }
+
+        context.Log.LogTrace("Searching memories for '{0}', collection '{1}', relevance '{2}'", text, collection, relevance);
```

Comment:
Logging the trace could generate lots of log, which is OK at LogTrace level.

But, do we care about potentially loggin PII here?

I'm just asking.  Not sure if it matters?

Reply:
Yeah, I'd certainly avoid the "text" parameter, as @glahaye says, to avoid potentially logging PII.

Reply:
Removed the text param